server/src/router/v1/index.ts

```diff
@@ -280,35 +518,287 @@ v1App.openapi(
+							newSubscription: subscribedEventSchema,
+						}),
+					},
+				},
+				description: 'Successful response',
+			},
+			[HttpStatusCodes.CONFLICT]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Conflict',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Forbidden',
+			},
+		},
+	}),
+	async (c) => {
+		const user = c.get('user');
+		const { eventID } = c.req.valid('param');
+
+		if (!user) {
+			return c.json({ error: 'Unauthorized' }, HttpStatusCodes.UNAUTHORIZED);
+		}
+
+		const newSubscription = await db
+			.insert(subscribedEvents)
+			.values({ userId: user.id, eventId: parseInt(eventID) })
+			.onConflictDoNothing()
+			.returning();
+
+		if (newSubscription.length === 0) {
+			return c.json({ error: 'Subscription already exists' }, HttpStatusCodes.CONFLICT);
+		}
+
+		return c.json({ newSubscription: newSubscription[0] }, HttpStatusCodes.OK);
+	},
+);
+
+v1App.openapi(
+	createRoute({
+		method: 'delete',
+		path: '/events/{eventID}/subscribe',
+		tags: ['events'],
+		summary: 'User unsubscribes to an event',
+		middleware: [authMiddleWare('user')],
+		request: {
+			params: eventIDSchema,
+		},
+		responses: {
+			[HttpStatusCodes.OK]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							deletedSubscription: subscribedEventSchema,
 						}),
 					},
 				},
 				description: 'Successful response',
 			},
+			[HttpStatusCodes.NOT_FOUND]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Not Found',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Forbidden',
+			},
 		},
 	}),
 	async (c) => {
+		const user = c.get('user');
 		const { eventID } = c.req.valid('param');
-		const eventSubscribers: User[] = await db
-			.select(getTableColumns(users))
-			.from(subscribedEvents)
-			.innerJoin(users, eq(users.id, subscribedEvents.userId))
-			.where(eq(subscribedEvents.eventId, parseInt(eventID)));
-		const formattedEventSubscribers = eventSubscribers.map(user => ({
-			...user,
-			createdAt: user.createdAt.toISOString(),
-		}));
-		return c.json({ eventSubscribers: formattedEventSubscribers }, HttpStatusCodes.OK);
+
+		if (!user) {
```

Comment:
unecessary

Reply:
shows a typescript error

'user' is possibly 'null'